pkg/oidc/verify.go

```diff
@@ -0,0 +1,70 @@
+package oidc
+
+import (
+	"context"
+	"fmt"
+	"github.com/coreos/go-oidc"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/logger"
+)
+
+// IDTokenVerifier Used to verify an ID Token and extends oidc.IDTokenVerifier from the underlying oidc library
+type IDTokenVerifier struct {
+	*oidc.IDTokenVerifier
+	*IDTokenVerificationOptions
+	allowedAudiences []string
```

Comment:
What about making this a `map[string]struct{}`, then there won't be any duplication and the lookup to check if the audience is allowed will be easier and more efficient